src/QsCompiler/Core/Dependencies.fs

```diff
@@ -39,58 +40,34 @@ type BuiltIn =
     /// Returns the set of callables that rewrite steps take dependencies on.
     /// These should be non-Generic callables only.
     static member RewriteStepDependencies =
-        ImmutableHashSet.Create(BuiltIn.RangeReverse.FullName, BuiltIn.Length.FullName)
+        ImmutableHashSet.Create(
+            BuiltIn.RangeReverse.FullName,
+            BuiltIn.Length.FullName,
+            BuiltIn.Inline.FullName,
+            BuiltIn.TargetInstruction.FullName,
+            BuiltIn.RequiresCapability.FullName,
+            BuiltIn.NoOp.FullName,
+            BuiltIn.ApplyConditionally.FullName,
+            BuiltIn.ApplyConditionallyA.FullName,
+            BuiltIn.ApplyConditionallyC.FullName,
+            BuiltIn.ApplyConditionallyCA.FullName,
+            BuiltIn.ApplyIfZero.FullName,
+            BuiltIn.ApplyIfZeroA.FullName,
+            BuiltIn.ApplyIfZeroC.FullName,
+            BuiltIn.ApplyIfZeroCA.FullName,
+            BuiltIn.ApplyIfOne.FullName,
+            BuiltIn.ApplyIfOneA.FullName,
+            BuiltIn.ApplyIfOneC.FullName,
+            BuiltIn.ApplyIfOneCA.FullName,
+            BuiltIn.ApplyIfElseR.FullName,
+            BuiltIn.ApplyIfElseRA.FullName,
+            BuiltIn.ApplyIfElseRC.FullName,
+            BuiltIn.ApplyIfElseRCA.FullName
+        )
 
     /// The set of all built in callables and attributes
-    static member AllBuiltIns =
```

Comment:
It seems to defeat the purpose of a built-in callable or attribute if it gets removed by the first rewrite step (syntax tree trimming) and is no longer available for any subsequent rewrite steps. Hence I think it would make the system more maintainable to save all of the built-ins through that rewrite step and not just the ones that we know to be used currently.

Reply:
Briefly summarizing our discussion:
Both the RewriteStepsDependencies and AllBuiltIns are deprecated and will be replaced by letting/requiring each rewrite step to declare its dependencies as part of the interface.